MetaMask/metamask-mobile
# feat: Enable zkSync Era in Swaps

## Explanation
This PR enables zkSync Era in Swaps.

Inspiration taken from this PR that enabled Avalanche: https://github.com/MetaMask/metamask-mobile/pull/3766/files

TODO:
- Fix an issue with unapproved ERC20 to ERC20 swaps (backend).

## Manual Testing Steps
- In the `node_modules/@metamask/swaps-controller/dist/constants.js` use this: `exports.DEV_BASE_URL = 'https://swap.dev-api.cx.metamask.io';`
- In the `node_modules/@metamask/swaps-controller/dist/swapsUtil.js` change the `getBaseApiURL` like this:
```
const getBaseApiURL = function (type, chainId) {
    const [apiChainId, apiBaseUrl] = chainId === constants_1.SWAPS_TESTNET_CHAIN_ID
        ? [constants_1.ETH_CHAIN_ID, constants_1.DEV_BASE_URL]
        : [chainId, constants_1.DEV_BASE_URL];
```
- Run the mobile app
- Add the zkSync Era network manually
- Check Swaps
